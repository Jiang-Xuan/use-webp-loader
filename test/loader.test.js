import webpack from './helpers/compiler'

describe('Loader', () => {
  test('Defaults', async () => {
    const config = {
      loader: {
        test: /(png|jpg|svg)/,
        options: {}
      }
    }

    const stats = await webpack('fixture.js', config)
    const module = stats.toJson().modules.filter(i => i.id === './hw-os-app@webp.png')
    const { source } = module[0]

    expect(source).toMatchSnapshot()
  })
})
