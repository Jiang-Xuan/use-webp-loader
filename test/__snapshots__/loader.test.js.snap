// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Defaults 1`] = `
"
        var isSupportWebp = require('is-support-webp')
        if (isSupportWebp()) {
          module.exports = __webpack_public_path__ + \\"4b5a9c4a10556d8f205e9f412d1662e6.webp\\"
        } else {
          module.exports = __webpack_public_path__ + \\"2af38ddf296b92cd6bbbf97322774c33.png\\"
        }
        "
`;
